### DISCUSSION TITLE
版本更新后潜艇电磁线圈失效/Submarine electromagnetic coil failure after version update

### DISCUSSION THREAD

**Discussion by dishfish** [FakeFishGames/Barotrauma#14104] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

版本更新后潜艇电磁线圈失效，之前可以正常工作，但是现在完全没有反应。我不知道它失效的原因。并且我发现原本用于侦测怪物并自动激活线圈的系统也失效了。/After the version update, the submarine electromagnetic coil failed. It used to work normally, but now there is no response at all. I don't know the reason for its failure. And I found that the system originally used to detect monsters and automatically activate coils also failed.

### Reproduction steps

[暴风之眼（sub） - 副本.zip](https://github.com/user-attachments/files/15751706/sub.-.zip)  （这是我的潜艇文件/This is my submarine file）
![持续放电](https://github.com/FakeFishGames/Barotrauma/assets/168256021/010732da-7225-4b70-8e7e-2a1c34257892)
（如上图所示，这种状态电磁线圈应该持续放电，但是没有任何反应/As shown in the above figure, the electromagnetic coil should continue to discharge in this state, but there is no response）
![结构图](https://github.com/FakeFishGames/Barotrauma/assets/168256021/48dd20a2-d21a-4665-91a0-8c338f9681e5)
（如上图所示，这是电磁线圈系统的结构图/As shown in the above figure, this is the structural diagram of the electromagnetic coil system）

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by jcx1991911** [comment]
电磁线圈必须直接连接电容器供电，与炮塔相同

**Reply by dishfish** [reply]
确实是这样，但是我还想请问一下，原先接在电磁线圈下面的那套自动侦测怪物触发线圈系统也失效了该怎么处理？
![a35bc45c398c4ae7c8070cb166f5dafe](https://github.com/FakeFishGames/Barotrauma/assets/168256021/2cd7723d-ac0e-4bee-9b90-bd0c96c4f154)

**Reply by dishfish** [reply]
就是原先有怪物靠近潜艇，对话框会弹出（内存组件）的消息然后触发线圈，但是现在完全没反应

**Reply by jcx1991911** [reply]
![20240609210227_1](https://github.com/FakeFishGames/Barotrauma/assets/116519004/87cd2bda-2d09-4311-b90b-0c3d1a23d7c1)
一切正常

**Reply by dishfish** [reply]
我在游戏里面测试了，内存组件这个东西应该在怪物来临时在聊天栏发出信息，但是现在没有反应，而且现在电磁线圈也要手动控制，那我之前的自动触发装置该怎么改进？

**Reply by jcx1991911** [reply]
如果你有使用Performance Fix的话我建议你检查一下设置或者停用Performance Fix

**Comment by Regalis11** [comment]
If google translate serves me right, the issue seems to have been resolved in the conversation above, closing.
